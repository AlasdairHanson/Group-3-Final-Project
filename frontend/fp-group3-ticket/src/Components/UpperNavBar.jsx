import React from "react";
import { Button, Nav, Navbar } from "react-bootstrap";
import { PersonFill } from "react-bootstrap-icons";
import { Link } from "react-router-dom";
import AddTicketButton from "./TicketPage/AddTicketButton";
const UpperNavBar = () => {
  return (
    <Navbar bg="light" expand="lg" sticky="top" className="navBar">
      <Navbar.Brand>
        <img
          src="https://i.imgur.com/Wa6jdim.png"
          alt="QA Logo"
          height="32px"
        />
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <Link to="/tickets">
            <Button className="ml-3 qaColButton"> Ticket Hub </Button>
          </Link>
          <Link to="/solutions">
            <Button variant="primary" className="ml-3 qaColButton">
              {" "}
              Solutions{" "}
            </Button>
          </Link>
          <Link to="/">
            <Button variant="primary" className="ml-3 qaColButton">
              <PersonFill /> Log Out{" "}
            </Button>
          </Link>
          <AddTicketButton />
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};
export default UpperNavBar;
