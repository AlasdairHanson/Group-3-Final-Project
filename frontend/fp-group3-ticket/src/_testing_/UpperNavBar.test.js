import { render, screen } from '@testing-library/react';
import {BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';
import UpperNavBar from '../Components/UpperNavBar';


  describe(`Test of UpperNavBar`, () => {

    test('Renders Ticket Hub Button', () => {
        render(<Router><UpperNavBar /></Router>);
        const linkElement = screen.getByText(/ticket hub/i);
        expect(linkElement).toBeInTheDocument();
      });

    test('Renders Solutions', () => {
        render(<Router><UpperNavBar /></Router>);
        const linkElement = screen.getByText(/solutions/i);
        expect(linkElement).toBeInTheDocument();
      });

    test('Renders Log Out', () => {
        render(<Router><UpperNavBar /></Router>);
        const linkElement = screen.getByText(/log out/i);
        expect(linkElement).toBeInTheDocument();
      });

    test(`UpperNavBar should match the snapshot every time it is rendered`, () => {
      const instance = create(<Router><UpperNavBar /></Router>).toJSON(); 
      expect(instance).toMatchSnapshot();
    });
  });