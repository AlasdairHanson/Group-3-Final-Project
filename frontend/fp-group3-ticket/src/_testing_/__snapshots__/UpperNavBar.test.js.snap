// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of UpperNavBar UpperNavBar should match the snapshot every time it is rendered 1`] = `
<nav
  className="navBar navbar navbar-expand-lg navbar-light bg-light sticky-top"
>
  <span
    className="navbar-brand"
  >
    <img
      alt="QA Logo"
      height="32px"
      src="https://i.imgur.com/Wa6jdim.png"
    />
  </span>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        href="/tickets"
        onClick={[Function]}
      >
        <button
          className="ml-3 qaColButton btn btn-primary"
          disabled={false}
          type="button"
        >
           Ticket Hub 
        </button>
      </a>
      <a
        href="/solutions"
        onClick={[Function]}
      >
        <button
          className="ml-3 qaColButton btn btn-primary"
          disabled={false}
          type="button"
        >
           
          Solutions
           
        </button>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="ml-3 qaColButton btn btn-primary"
          disabled={false}
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fillRule="evenodd"
            />
          </svg>
           Log Out
           
        </button>
      </a>
      <button
        className="ml-3 qaColButton btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8.5 4.5a.5.5 0 0 0-1 0v3h-3a.5.5 0 0 0 0 1h3v3a.5.5 0 0 0 1 0v-3h3a.5.5 0 0 0 0-1h-3v-3z"
            fillRule="evenodd"
          />
        </svg>
         New Ticket
      </button>
    </div>
  </div>
</nav>
`;
